pkgs/development/libraries/linenoise/default.nix

```diff
@@ -1,29 +1,48 @@
-{ lib, stdenv, fetchFromGitHub }:
+{ lib
+, stdenv
+, fetchFromGitHub
+, validatePkgConfig
+}:
 
 stdenv.mkDerivation {
   pname = "linenoise";
-  version = "1.0.10";  # Its version 1.0 plus 10 commits
+  version = "1.0-34-g93b2db9";
 
   src = fetchFromGitHub {
     owner = "antirez";
     repo = "linenoise";
-    rev = "c894b9e59f02203dbe4e2be657572cf88c4230c3";
-    sha256 = "0wasql7ph5g473zxhc2z47z3pjp42q0dsn4gpijwzbxawid71b4w";
+    rev = "1.0-34-g93b2db9";
+    hash = "sha256-GsrYg16gpjHkkmpCU3yGzqNS/buZl+JoWALLvwzmT4A=";
   };
 
-  buildPhase = ./create-pkg-config-file.sh;
+  nativeBuildInputs = [ validatePkgConfig ];
+
+  buildPhase = ''
+    runHook preBuild
+
+    $CC -c -gctf -gdwarf -o linenoise.o linenoise.c
+    $CC -shared -o liblinenoise.so linenoise.o
+    $AR rcs liblinenoise.a linenoise.o
+
+    runHook postBuild
+  '';
 
   installPhase = ''
-    mkdir -p $out/{lib/pkgconfig,src,include}
-    cp linenoise.c $out/src/
-    cp linenoise.h $out/include/
-    cp linenoise.pc $out/lib/pkgconfig/
+    runHook preInstall
+
+    install -d $out/lib/pkgconfig $out/include
+    install -m644 linenoise.h     $out/include/
+    install -m644 liblinenoise.a  $out/lib/
+    install -m644 liblinenoise.so $out/lib/
+    substituteAll ${./linenoise.pc.in} $out/lib/pkgconfig/linenoise.pc
+
+    runHook postInstall
   '';
 
   meta = {
     homepage = "https://github.com/antirez/linenoise";
     description = "A minimal, zero-config, BSD licensed, readline replacement";
-    maintainers = with lib.maintainers; [ fstamour ];
+    maintainers = with lib.maintainers; [ fstamour remexre ];
     platforms = lib.platforms.unix;
```

Comment:
if this is supposed to build on darwin mark darwin as broken since the build currently fails in ci, if it is only supposed to build on linux set platforms to linux and if it's supposed to build on all unix-like platforms except for darwin set `badPlatforms = lib.platforms.darwin;`